### PR TITLE
add load and save to docker

### PR DESCRIPTION
**Pull request by anandkumarpatel**

### CODE REVIEW

**Comment by bkendall** [comment]
@anandkumarpatel merge conflict

File: lib/middleware.js [hunk 1]
```diff
@@ -496,6 +497,18 @@ var middlewares = module.exports = {
...
+    res.setHeader('Content-Type', 'application/x-tar');
+    res.status(200);
+    stringStream.pipe(res);
+  },
+  imageLoad: function (req, res) {
+    req.on('data', noop);
+    req.on('end', function () {
+      res.sendStatus(200);
+    });
+  },
```
Comment (bkendall) [inline lib/middleware.js]: could we not have save/load actually use the images to send _something_ that is accurate and useful? Additionally, the `findImage` middleware should be run prior to `imageSave` to validate the image actually exists in the mock (not _always_ say that the image exists)

Comment (anandkumarpatel) [inline lib/middleware.js]: updated to use real busy box tar (smallest image i could find, 2mb) now we also load this into the image array :)

File: test/docker.js [hunk 2]
```diff
@@ -478,6 +480,26 @@ describe('images', function () {
...
+    it('should save an image', function (done) {
+      docker.getImage('testImage').get(handleStream(done));
+    });
+    it('should load an image', function (done) {
+      var imageStream = fs.createReadStream('misc/busybox.tar');
+      docker.loadImage(imageStream, function (err) {
+        if (err) { return done(err); }
+        docker.listImages(function (err, images) {
+          if (err) { return done(err); }
+          expect(images).to.have.length(5);
```
Comment (bkendall) [inline test/docker.js]: you're expecting `5` here - how do you know it's `5`? how do you know the one _you added_ is in that list?

Comment (anandkumarpatel) [inline test/docker.js]: I just counted how many images should be here at this point. Does make it a bit less flexible. Will has a check just for the specific image

File: lib/middleware.js [hunk 3]
```diff
@@ -309,6 +311,12 @@ var middlewares = module.exports = {
   },
   respondImages: function (req, res) {
     var data = Object.keys(images).map(function (id) {
+      // if we have full image info, we can return that instead
+      // assume if container_config exist on image, it is full object
+      if (images[id].container_config) {
+        return images[id];
+      }
+      // fake return object if we do not have all info
```
Comment (bkendall) [inline lib/middleware.js]: I didn't see this before. I don't think this will include `RepoTags` if you take this short cut. is that true?

Comment (anandkumarpatel) [inline lib/middleware.js]: repo tags set here 
https://github.com/Runnable/docker-mock/pull/39/files#diff-d7dfcb350720309421bbb465055aafdfR522

it should pick up repo-tags container_config should only exist if we are using real docker image format

**Comment by anandkumarpatel** [comment]
on a side note: would you mind if I added an image to the image pool when we create image-builder in create route (POST /containers/create) ? in that route we can check if `Image=runnable%2Fimage-builder` and add value of `RUNNABLE_DOCKERTAG` to  the image object we store.

**Comment by bkendall** [comment]
@anandkumarpatel yes, I mind

**Comment by bkendall** [comment]
LGTM otherwise, any last thoughts, @anandkumarpatel

**Pull request merged**